RubberduckTests/Inspections/IsMissingWithNonArgumentParameterInspectionTests.cs

```diff
@@ -0,0 +1,186 @@
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void IsMissingWithNonArgumentParameter_DoesNotReportWhenIgnored()
+        {
+            const string inputCode =
+                @"
+Public Sub Foo(bar As Variant)
+'@Ignore IsMissingWithNonArgumentParameter
+    Debug.Print IsMissing(42)
+End Sub
+";
+
+            const int expected = 0;
+            var actual = ArrangeAndGetInspectionCount(inputCode);
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void IsMissingWithNonArgumentParameter_DoesNotReportOptionalVariantArgument()
+        {
+            const string inputCode =
+                @"
+Public Sub Foo(Optional bar As Variant)
+    Debug.Print IsMissing(bar)
+End Sub
+";
+
+            const int expected = 0;
+            var actual = ArrangeAndGetInspectionCount(inputCode);
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void IsMissingWithNonArgumentParameter_DoesNotReportParamArray()
+        {
+            const string inputCode =
+                @"
+Public Sub Foo(ParamArray bar() As Variant)
+    Debug.Print IsMissing(bar)
+End Sub
+";
+
+            const int expected = 0;
+            var actual = ArrangeAndGetInspectionCount(inputCode);
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void IsMissingWithNonArgumentParameter_ReportsArgumentAsParameterInExpression()
+        {
+            const string inputCode =
+                @"
+Public Sub Foo(Optional bar As Variant)
+    Debug.Print IsMissing(Baz(bar))
+End Sub
+
+Public Function Baz(arg As Variant) As Variant
+End Function
+";
+
+            const int expected = 1;
+            var actual = ArrangeAndGetInspectionCount(inputCode);
+
+            Assert.AreEqual(expected, actual);
+        }
+
+        private int ArrangeAndGetInspectionCount(string code)
+        {
+            var builder = new MockVbeBuilder();
+            var project = builder.ProjectBuilder("TestProject1", "TestProject1", ProjectProtection.Unprotected)
+                .AddComponent("Module1", ComponentType.StandardModule, code)
+                .AddReference("VBA", MockVbeBuilder.LibraryPathVBA, 4, 2, true)
+                .Build();
+            var vbe = builder.AddProject(project).Build();
+
+
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new IsMissingWithNonArgumentParameterInspection(state);
+                var inspectionResults = inspection.GetInspectionResults(CancellationToken.None);
+
+                return inspectionResults.Count();
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void InspectionName()
```

Comment:
this test should already be covered [here](https://github.com/rubberduck-vba/Rubberduck/blob/next/RubberduckTests/Inspections/GeneralInspectionTests.cs#L44)

Reply:
Removed (in this PR).  I'll delete the others as I run across the.